### ISSUE TITLE
Float overloads for Drag functions

### ISSUE THREAD

**Issue by bear24rw** [epezent/implot#120]
It would be helpful to templatize the Drag functions to also allow passing a float. This would allow the value to more easily iteropt with other imgui widgets. In my case I also allow controlling the value via ImGui::DragFloat which only takes floats.

**Comment by epezent** [comment]
`ImGui::DragScalar` can take doubles, though I see the value in supporting `float` and the other supported types for sure.

**Comment by bear24rw** [comment]
> ImGui::DragScalar can take doubles

Oh nice, I didn't know about that function.

> though I see the value in supporting float and the other supported types for sure.

👍🏻

**Comment by epezent** [comment]
So, I just finished templatizing these functions and was about to merge them, but I'm reconsidering it now. 

I'm afraid this will take us down the rabbit hole of templatizing _every_ API function that takes `double`. We have to draw the line somewhere, and ask users to do some conversions on their own. Obviously it's unreasonable to expect users to convert their plot data, but for utility functions like these I don't think it's too much to ask.

As I said, ImGui supports `double` for most widgets via `ImGui::XXXScalar`. We _might_ consider doing [something like I've done before](https://github.com/mahilab/mahi-gui/blob/883e1461937b984d653f2d4b365b1743dfda8c02/include/Mahi/Gui/imgui_custom.hpp#L54-61), exposing wrappers such as `SliderDouble`. I'm not sure if this is a great idea though. If we did it, we'd want to at least make it a compile time option via macro. 

If this continues to be a common concern, we will need to think very hard about moving to a real, header-only templated API.
